src/main/java/org/openrewrite/java/testing/jmockit/JMockitMockUpToMockito.java

```diff
@@ -0,0 +1,537 @@
+import org.openrewrite.*;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.JavaParser;
+import org.openrewrite.java.JavaTemplate;
+import org.openrewrite.java.search.UsesType;
+import org.openrewrite.java.tree.*;
+import org.openrewrite.marker.SearchResult;
+import org.openrewrite.staticanalysis.RemoveUnusedLocalVariables;
+
+import java.lang.reflect.Method;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.Collectors.toMap;
+import static org.openrewrite.java.testing.jmockit.JMockitBlockType.MockUp;
+import static org.openrewrite.java.testing.mockito.MockitoUtils.maybeAddMethodWithAnnotation;
+import static org.openrewrite.java.tree.Flag.Private;
+import static org.openrewrite.java.tree.Flag.Static;
+
+public class JMockitMockUpToMockito extends Recipe {
+    private static final String MOCKITO_CLASSPATH = "mockito-core-5";
+    private static final String MOCKITO_ALL_IMPORT = "org.mockito.Mockito.*";
+    private static final String MOCKITO_MATCHER_IMPORT = "org.mockito.ArgumentMatchers.*";
+    private static final String MOCKITO_DELEGATEANSWER_IMPORT = "org.mockito.AdditionalAnswers.delegatesTo";
+    private static final String JMOCKIT_MOCKUP_IMPORT = MockUp.getFqn();
+    private static final String JMOCKIT_MOCK_IMPORT = "mockit.Mock";
+    private static final String MOCKITO_STATIC_PREFIX = "mockStatic";
+    private static final String MOCKITO_STATIC_IMPORT = "org.mockito.MockedStatic";
+    private static final String MOCKITO_MOCK_PREFIX = "mock";
+    private static final String MOCKITO_CONSTRUCTION_PREFIX = "mockCons";
+    private static final String MOCKITO_CONSTRUCTION_IMPORT = "org.mockito.MockedConstruction";
+
+    @Override
+    public String getDisplayName() {
+        return "Rewrite JMockit MockUp to Mockito statements";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Rewrites JMockit `MockUp` blocks to Mockito statements. This recipe will not rewrite private methods in MockUp.";
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return Preconditions.check(new UsesType<>(MockUp.getFqn(), false), new JMockitMockUpToMockitoVisitor());
+    }
+
+    private static class JMockitMockUpToMockitoVisitor extends JavaIsoVisitor<ExecutionContext> {
+        private final Map<String, J.Identifier> tearDownMocks = new HashMap<>();
+
+        /**
+         * Handle at class level because need to handle the case where when there is a MockUp in a setup method, and we
+         * need to close the migrated mockCons in the teardown, yet the teardown method comes before the setup method
+         */
+        @Override
+        public J.ClassDeclaration visitClassDeclaration(J.ClassDeclaration classDecl, ExecutionContext ctx) {
+            // Handle @Before/@BeforeEach mockUp
+            Set<J.MethodDeclaration> mds = TreeVisitor.collect(
```

Comment:
TreeVisitor.collect is marked @Incubating and is an experimental feature and may have breaking changes, is it possible to avoid this one?

Reply:
I think the final solution's code will very likely the same as TreeVisitor.collect()'s.